opm/autodiff/BlackoilModelBase_impl.hpp

```diff
@@ -1124,6 +1217,91 @@ namespace detail {
 
 
 
+
+    template <class Grid, class Implementation>
+    void BlackoilModelBase<Grid, Implementation>::solveWellEq(const std::vector<ADB>& mob_perfcells,
+                                                              const std::vector<ADB>& b_perfcells,
+                                                              SolutionState& state,
+                                                              WellState& well_state)
+    {
+        V aliveWells;
+        const int np = wells().number_of_phases;
+        std::vector<ADB> cq_s(np, ADB::null());
+        std::vector<int> indices = variableWellStateIndices();
+        SolutionState state0 = state;
+        asImpl().makeConstantState(state0);
+
+        std::vector<ADB> mob_perfcells_const(np, ADB::null());
+        std::vector<ADB> b_perfcells_const(np, ADB::null());
+        for (int phase = 0; phase < np; ++phase) {
+            mob_perfcells_const[phase] = ADB::constant(mob_perfcells[phase].value());
+            b_perfcells_const[phase] = ADB::constant(b_perfcells[phase].value());
+        }
+
+        int it  = 0;
+        bool converged;
+        do {
+            // bhp and Q for the wells
+            std::vector<V> vars0;
+            vars0.reserve(2);
+            variableWellStateInitials(well_state, vars0);
+            std::vector<ADB> vars = ADB::variables(vars0);
+
+            SolutionState wellSolutionState = state0;
+            variableStateExtractWellsVars(indices, vars, wellSolutionState);
+            asImpl().addWellEq(wellSolutionState, well_state, mob_perfcells_const, b_perfcells_const, aliveWells, cq_s);
+            addWellControlEq(wellSolutionState, well_state, aliveWells);
+            converged = getWellConvergence(it);
+
+            if (converged) {
+                break;
+            }
+
+            ++it;
+            std::vector<ADB> eqs;
+            eqs.reserve(2);
+            eqs.push_back(residual_.well_flux_eq);
+            eqs.push_back(residual_.well_eq);
+            ADB total_residual = vertcatCollapseJacs(eqs);
+            const std::vector<M>& Jn = total_residual.derivative();
+            const Eigen::SparseLU< M > solver(Jn[0]);
```

Comment:
> Adding this warning to the ignored list [...] would suppress it

First, that's a **really** bad idea.  We want to be aware of "set-but-not-used".  Second, I don't see how the suppression mechanism can possibly work in this case.  The context of `info` (at least in Eigen 3.2.1) is
```C++
template<typename Index>
class COLAMDOrdering
{
  public:
    typedef PermutationMatrix<Dynamic, Dynamic, Index> PermutationType;
    typedef Matrix<Index, Dynamic, 1> IndexVector;

    /** Compute the permutation vector form a sparse matrix */
    template <typename MatrixType>
    void operator() (const MatrixType& mat, PermutationType& perm)
    {
      Index m = mat.rows();
      Index n = mat.cols();
      Index nnz = mat.nonZeros();
      // Get the recommended value of Alen to be used by colamd
      Index Alen = internal::colamd_recommended(nnz, m, n);
      // Set the default parameters
      double knobs [COLAMD_KNOBS];
      Index stats [COLAMD_STATS];
      internal::colamd_set_defaults(knobs);

      Index info;
      IndexVector p(n+1), A(Alen);
      for(Index i=0; i <= n; i++)   p(i) = mat.outerIndexPtr()[i];
      for(Index i=0; i < nnz; i++)  A(i) = mat.innerIndexPtr()[i];
      // Call Colamd routine to compute the ordering 
      info = internal::colamd(m, n, Alen, A.data(), p.data(), knobs, stats);
      eigen_assert( info && "COLAMD failed " );

      perm.resize(n);
      for (Index i = 0; i < n; i++) perm.indices()(p(i)) = i;
    }
};
```
so the class is a template.  In other words, we would in general have to guard every call site to suppress the warning.